### ISSUE TITLE
Docker without Host Network

### ISSUE THREAD

**Issue by linedpaper** [xteve-project/xTeVe#5]
When running in a container and mapping only port 34400 (not using "Host" networking), xteve thinks it's IP is the Docker IP.  When trying to download a backup I am sent to a URL of the Docker IP, however a manual edit to the host IP allows download of backup.

**Comment by mar-mei** [comment]
xTeVe uses the network stack and all IP addresses provided by the OS. I do not know exactly how Docker works, but for me that sounds like a NAT problem between the host IP and the IP from the Docker image.

**Comment by MCTyler** [comment]
I think OP is asking to be able to configure xTeVe where to send the backup file as a selection in settings. Feature Request.

**Comment by linedpaper** [comment]
That's seems more like it.  A lot of Docker apps allow the manual
specification of a hostname or ip.

**Comment by xteve-project** [comment]
As I said, I'm not familiar with Docker, I use FreeBSD Jails to isolate programs.
I do not know any programs, in which one indicates which IP address these have. I know only programs, in which one can specify, which IP addresses may call this program.

xTeVe sets the requested IP address / hostname dynamically, since they must also be in the M3U or DVR Lineup.

My server has several IP addresses. Here's an example:
```
...
2019/08/06 15:19:44 [xTeVe] System IP Addresses:   IPv4: 2 | IPv6: 7
2019/08/06 15:19:44 [xTeVe] Hostname:              mbp.local
...
```

Request over IP: 192.168.178.40:34400/m3u/xteve.m3u
For the xteve.m3u the **IP address** will be adjusted
```
#EXTINF:0 channelID="x-ID.2" tvg-chno="1002" tvg-name="M3U8 Test" tvg-id="1002" tvg-logo="" group-title="Buffer 1",M3U8 Test
http://192.168.178.3:34400/stream/f39126c7469d3640bc13df03c9fa71fd
```

Request over IP: 127.0.0.1:34400/m3u/xteve.m3u
For the xteve.m3u the **IP address** will be adjusted:
```
#EXTINF:0 channelID="x-ID.2" tvg-chno="1002" tvg-name="M3U8 Test" tvg-id="1002" tvg-logo="" group-title="Buffer 1",M3U8 Test
http://127.0.0.1:34400/stream/f39126c7469d3640bc13df03c9fa71fd
```

Request over hostname: mbp.local:34400/m3u/xteve.m3u
For the xteve.m3u the **hostname** will be adjusted:
```
#EXTINF:0 channelID="x-ID.2" tvg-chno="1002" tvg-name="M3U8 Test" tvg-id="1002" tvg-logo="" group-title="Buffer 1",M3U8 Test
http://mbp.local:34400/stream/f39126c7469d3640bc13df03c9fa71fd
```

When an M3U client loads xteve.m3u over a public domain (http://example.com:34400/m3u/xteve.m3u), the client will get back the M3U with the domain in the streaming URL. If parallel Plex queried the DVR Linup over localhost, Plex got the lineup with localhost addresses.

**Comment by linedpaper** [comment]
Setting the Ip or hostname only applies and refers to links within the site
itself refer back to the specified hostname or IP instead of what it thinks
it is itself.  A few examples are: Unifi Controller, Ombi and Tautulli.

**Comment by xteve-project** [comment]
Right, in the Unifi controller, the hostname can be changed. But for some other reason, the controller is usually the DNS server on the network and also configures the USG.

xTeVe is not a DNS server, xTeVe is a web server and hunches all data from the HTTP protocol:

Here's a quote on xTeVe with hostname: **map.local**:

Here is the HTTP header

&{GET /web/img/log.png HTTP/1.1 1 1 map[Accept:[image/webp,image/apng,image/*,*/*;q=0.8] Accept-Encoding:[gzip, deflate] Accept-Language:[de-DE,de;q=0.9,en-US;q=0.8,en;q=0.7] Connection:[keep-alive] Referer:[**http://mbp.local:34400/web/**] User-Agent:[Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3770.142 Safari/537.36]] {} <nil> 0 [] false mbp.local:34400 map[] map[] <nil> map[] 127.0.0.1:56737 /web/img/log.png <nil> <nil> <nil> 0xc0158267c0}

The host name comes from the HTTP header, because everything is in there what an HTTP client / server needs to know.

![image](https://user-images.githubusercontent.com/53275662/62579363-d721b000-b8a3-11e9-867e-e894e9dd5201.png)

![image](https://user-images.githubusercontent.com/53275662/62579421-fe787d00-b8a3-11e9-9077-a059b7124eb3.png)

I do not have a Docker that's why I can not check it

**Comment by kudos** [comment]
Works fine for me in Docker. The only place I've seen it use the internal Docker IP is in the log output, like:

```
2019/08/07 11:24:47 [xTeVe] Web Interface:         http://172.20.0.1:8082/web/
```

**Comment by linedpaper** [comment]
Can you try downloading a copy of the xteve backup?  That's the other
location I see xteve using the Docker IP instead of host.

**Comment by mar-mei** [comment]
The problem with the URL is fixed in v2.0.1. That xTeVe sees the host IP must be at your docker configuration.
